zzre/game/systems/dialog/DialogTrading.cs

```diff
@@ -0,0 +1,245 @@
+        DefaultEcs.Entity entity,
+        components.ui.DialogTrading trading,
+        int index,
+        ItemRow currency,
+        int price,
+        UID uid,
+        Rect bgRect)
+    {
+        var cards = db.Items.OrderBy(itemRow => itemRow.CardId.EntityId).ToArray();
+        var card = db.GetItem(uid);
+
+        var purchase = new components.ui.ElementId(index);
+        var offset = bgRect.Center + new Vector2(-205, -120 + 55 * index);
+
+        preload.CreateImage(entity)
+            .With(offset)
+            .With(preload.Itm000, card.CardId.EntityId)
+            .Build();
+
+        preload.CreateLabel(entity)
+            .With(offset + new Vector2(50, 16))
+            .WithText(card.Name)
+            .With(preload.Fnt002)
+            .Build();
+
+        preload.CreateLabel(entity)
+            .With(offset + new Vector2(325, 12))
+            .WithText($"{{0*x}}{price}")
+            .With(preload.Fnt000)
+            .Build();
+
+        var inventory = zanzarah.CurrentGame!.PlayerEntity.Get<Inventory>();
+        if (inventory.CountCards(currency.CardId) >= price) {
+            preload.CreateButton(entity)
+                .With(purchase)
+                .With(offset + new Vector2(365, 0))
+                .With(new components.ui.ButtonTiles(20, 21))
+                .With(preload.Btn001)
+                .Build();
+        } else {
+            preload.CreateImage(entity)
+                .With(offset + new Vector2(365, 0))
+                .With(preload.Btn001, 22)
+                .Build();
+        }
+
+        trading.CardPurchaseButtons[purchase] = card;
+    }
+
+    private const float ButtonOffsetY = -50f;
+    private void CreateSingleButton(DefaultEcs.Entity entity, UID textUID, components.ui.ElementId elementId, Rect bgRect)
+    {
+        preload.CreateButton(entity)
+            .With(elementId)
+            .With(new Vector2(bgRect.Center.X, bgRect.Max.Y + ButtonOffsetY))
+            .With(new components.ui.ButtonTiles(0, 1))
+            .With(components.ui.FullAlignment.TopCenter)
+            .With(preload.Btn000)
+            .WithLabel()
+            .With(preload.Fnt000)
+            .WithText(textUID)
+            .Build();
+
+        // TODO: Set cursor position in dialog trading
+    }
+    private void HandleElementDown(DefaultEcs.Entity entity, components.ui.ElementId clickedId)
+    {
+        var uiEntity = Set.GetEntities()[0];
+        ref var trading = ref uiEntity.Get<components.ui.DialogTrading>();
+
+        if (trading.CardPurchaseButtons.TryGetValue(clickedId, out var card)) {
+            trading.Profile.Dispose();
+            trading.Purchase = card;
+            trading.Profile = CreateItemProfile(uiEntity, trading, card);
+        }
+        else if (clickedId == IDYes) {
+            var purchase = trading.Purchase!;
+            var price = trading.CardTrades.First(((int price, UID uid) trade) => db.GetItem(trade.uid).Name == purchase.Name).Item1;
```

Comment:
You can shorten `((int price, UID uid) trade)` to just trade.
Comparing names is more error-prone and not necessary, the UID is meant to compare identities
Instead of `Item1` you can use the given name.
So this line can just be `.First(trade => trade.uid == purchase.Uid).price`